src/res_printer.ml

```diff
@@ -99,6 +99,14 @@ let hasCommentBelow tbl loc =
   | [] -> false
   | exception Not_found -> false
 
+let hasNestedJsx expr =
+  match expr.Parsetree.pexp_desc with
+  | Pexp_construct
+      ({txt = Longident.Lident "::"}, Some {pexp_desc = Pexp_tuple (hd :: [_])})
+    ->
+    ParsetreeViewer.isJsxExpression hd
```

Comment:
What about some children are, and some are not, JSX expressions?

Reply:
Perhaps a good test to add.

Reply:
Yes, I don't think this is correct without recursion. If hd is not a JSX expression, we still need to recurse over tail.

E.g. for cases like

```rescript
<A>
  {children}
  <B/>
<A/>
```

Reply:
Ouch, sure. I’ve recovered the recursion add added the tests